controllers/metallb_controller.go

```diff
@@ -149,15 +158,36 @@ func (r *MetalLBReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 
+	if r.EnvConfig.IsOpenshift {
+		return ctrl.NewControllerManagedBy(mgr).
+			For(&metallbv1beta1.MetalLB{}).
+			Watches(&openshiftapiv1.Network{}, &handler.EnqueueRequestForObject{}).
+			Complete(r)
+	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&metallbv1beta1.MetalLB{}).
 		Complete(r)
 }
 
-func (r *MetalLBReconciler) syncMetalLBResources(config *metallbv1beta1.MetalLB) error {
+func (r *MetalLBReconciler) syncMetalLBResources(ctx context.Context, config *metallbv1beta1.MetalLB) error {
 	logger := r.Log.WithName("syncMetalLBResources")
 	logger.Info("Start")
 
+	bgpType := params.BGPType(config, r.EnvConfig.IsOpenshift)
+	if r.EnvConfig.MustDeployFRRK8sFromCNO && r.EnvConfig.IsOpenshift && (bgpType == metallbv1beta1.FRRK8sExternalMode) {
```

Comment:
not sure I understand the `MustDeployFRRK8sFromCNO` env var, I thought "external" in openshift always comes from cno?